specs/exp-proof.md

```diff
@@ -71,6 +73,11 @@ The exponentiation circuit consists of the following columns:
     - Both multiplicand's of the `mul_gadget`, i.e. `a` and `b` MUST equal `base` of the exponentiation operation, that is:
         - `mul_gadget.a == base`
         - `mul_gadget.b == base` (for both these cases we equate each 64-bit limb)
+- For every row where `padding == true` and `is_final == false`, validate that:
+    - next row is also `padding == true`
+    - all other columns are set to 0
```

Comment:
I think when `padding == true` and `is_final == true` we need a condition that checks that there's one more row with the same values. And then we can finish. 
Otherwise, you still have an algorithm that cannot terminate.